### DISCUSSION TITLE
Extending ToastRootProps in vite library mode throws typescript error

### DISCUSSION THREAD

**Discussion by hrubysi** [unovue/reka-ui#946] [Q&A] [answered]
### Link to minimal reproduction
https://stackblitz.com/edit/vitejs-vite-ur52p7

### Steps to reproduce
`npm run build`

### The problem
Extending ToastRootProps in useToast.ts throws type error. The code is based on shad cn toast.

`TS2742: The inferred type of 'useToast' cannot be named without a reference to '../node_modules/radix-vue/dist/Primitive/Primitive.js'. This is likely not portable. A type annotation is necessary.`

Also definig emits as ToastRootEmits in Toast.vue throws type error.

`TS2742: The inferred type of 'default' cannot be named without a reference to '../node_modules/radix-vue/dist/Toast/utils.js'. This is likely not portable. A type annotation is necessary.`

I'm not sure if this problem is related to radix-vue, but would anyone know of some solution for this?

**Comment by hrubysimon** [comment]
I managed to solve the problem by explicitly mapping paths in my tsconfig.
The question reamains, is this caused by radix-vue? @zernonia 

```
"paths": {
  "radix-vue/dist/Toast/utils": ["node_modules/radix-vue/dist/Toast/utils"],
  "radix-vue/dist/Primitive": ["node_modules/radix-vue/dist/Primitive"],
}
```

**Reply by zernonia** [reply]
@hrubysimon  there's a similar issue being logged, and it should be fixed now

https://github.com/radix-vue/radix-vue/issues/950

**Reply by hrubysi** [reply]
I confirm, that the first error is fixed now. For the second, I created a new [issue](https://github.com/radix-vue/radix-vue/issues/960) .

**Comment by hrubysi** [comment, marked as answer]
I found two solutions for this:

1. Adding `baseUrl: "."` to tsconfig
2. Changing `nodeResolution` from `bundler` to `node`

By reading ts docs, none of the two should really be used, so that leaves me confused.
